examples/fluids/src/dm_utils.c

```diff
@@ -313,11 +457,12 @@ PetscErrorCode DMSetupByOrderBegin_FEM(PetscBool setup_faces, PetscBool setup_co
     PetscCall(PetscDualSpaceGetOrder(fe_coord_dual_space, &fe_coord_order));
 
     // Create FE for coordinates
-    PetscCheck(fe_coord_order == 1 || coord_order == 1, PetscObjectComm((PetscObject)dm), PETSC_ERR_USER_INPUT,
+    PetscCheck(fe_coord_order == 1 || coord_order == 1, comm, PETSC_ERR_USER_INPUT,
                "Only linear mesh geometry supported. Recieved %d order", fe_coord_order);
     PetscCall(PetscFECreateLagrange(comm, dim, num_comp_coord, is_simplex, fe_coord_order, q_order, &fe_coord_new));
     if (setup_faces) PetscCall(PetscFEGetHeightSubspace(fe_coord_new, 1, &fe_coord_face_new));
     PetscCall(DMProjectCoordinates(dm, fe_coord_new));
+    PetscCall(DMLocalizeCoordinates(dm));  // Update CellCoordinateDM with projected coordinates
```

Comment:
I'm not sure how necessary this is, primarily because we don't have any tests that actually change the FE coordinate space. But I think this is required to get the cell geometry correct. And I believe that this should still respect `-dm_sparse_localize` as well.

Reply:
I'll go ahead and call this good since we can't really test it right now. Either including it is wrong and we'll find out when we get there, or not including it is right... and we'll still find that out later anyways.